Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -2937,7 +2938,10 @@ int FrameLoader::numPendingOrLoadingRequests(bool recurse) const
 String FrameLoader::userAgent(const URL& url) const
 {
     String userAgent;
-    if (RefPtr localFrame = dynamicDowncast<LocalFrame>(m_frame->mainFrame())) {
+    if (auto quirk = Quirks::storageAccessUserAgentStringQuirkForDomain(url); !quirk.isEmpty())
```

Comment:
This looks like it will quirk the UA for any URL which matches a domain on our list, so nothing is really restricting this to the storage access API. Is that okay?

Reply:
Good question. I think it should be okay because if we know that we need a UA quirk for a site to work, even after quirking the storage access prompt, and we decide that using the UA quirk is safe in that case, then using the UA quirk without the storage access prompt quirk should be safe, too. We could couple these more tightly, but I'll leave that to a follow up patch if we decide that's important.